Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -306,6 +380,22 @@ void __stdcall PlayerLaunch_AFTER(unsigned int ship, unsigned int client)
 		mobiledockClients.erase(client);
 	}
 
+	// Land ship to proxy base in carrier's system if they are in different systems.
+	if (mobiledockClients[client].carrierSystem == -1)
+	{
+		string scProxyBase = HkGetPlayerSystemS(HkGetClientIdFromCharname(mobiledockClients[client].wscDockedWithCharname)) + "_proxy_base";
+		uint iBaseID;
+		if (pub::GetBaseID(iBaseID, scProxyBase.c_str()) == -4)
+		{
+			PrintUserCmdText(client, L"No proxy base in system detected. Contact a developer about this please.");
+			return;
+		}
+
+		pub::Player::ForceLand(client, iBaseID);
+		PrintUserCmdText(client, L"Navmap updated succesfully.");

```

Comment:
seems like a bizarre message?

Reply:
Not more bizarre if player tries to undock and sees that he is still at proxy base without any explanation.